app/js/map/map.controller.js

```diff
@@ -0,0 +1,94 @@
+(function() {
+  'use strict';
+  angular.module('upcomingStops.map').controller('MapController', MapController);
+
+  MapController.$inject = ['$scope'];
+
+  function MapController($scope) {
+    var vm = this;
+    vm.map = null;
+
+    vm.latestLocation = { lat: 61.498180, lng: 23.762195 };
+
+    vm.showUserPosition = true;
+    vm.userPosMarker = null;
+
+    vm.setMap = setMap;
+    vm.getMap = getMap;
+    vm.centerToUser = centerToUser;
+
+    function setMap(map) {
+      vm.map = map;
+      initUserPositionMarker(vm.map, vm.latestLocation);
+
+      $scope.$emit('mapReady', vm.map);
+    }
+
+    function getMap() {
+      return vm.map;
+    }
+
+    function initUserPositionMarker(map, initPosition) {
+      vm.userPosMarker = map.createMarker({
+        position: initPosition,
+        icon: 'img/userpositionmarker.png'
+      });
+      vm.userPosMarker.setClickable(false);
+
+      if (vm.showUserPosition) {
+        showUserPosMarker();
+      }
+    }
+
+    function updateUserPosMarker() {
+      if (vm.userPosMarker !== null) {
+        vm.userPosMarker.setPosition(vm.latestLocation);
+      }
+    }
+
+    function showUserPosMarker() {
+      if (vm.map !== null && vm.latestLocation != null) {
+        if (vm.userPosMarker === null) {
```

Comment:
Read the whole thing, the outer block continues after the inner one. Though it can probably be changed so that the userPosMarker is initialized automatically but not attached to the map, and thus make the inner block obsolete.